File: Bai_5/main.py
def fibonacci(n):
    if n <= 1:
        return n
    return fibonacci(n - 1) + fibonacci(n - 2)

# Hàm để tìm các số Fibonacci trong dãy số đã nhập
def find_fibonacci_in_sequence(sequence):
    fibonacci_numbers = []
    for num in sequence:
        # Kiểm tra nếu số trong dãy là số Fibonacci
        i = 0
        while fibonacci(i) <= num:
            if fibonacci(i) == num:
                fibonacci_numbers.append(num)
                break
            i += 1
    return fibonacci_numbers

# Nhập dãy số từ người dùng
user_input = input("Nhập vào dãy số cách nhau bằng dấu cách: ")
numbers = list(map(int, user_input.split()))

# Tìm và in ra các số Fibonacci trong dãy số đã nhập
fibonacci_in_sequence = find_fibonacci_in_sequence(numbers)
if fibonacci_in_sequence:
    print("Các số Fibonacci trong dãy số đã nhập là:")
    print(fibonacci_in_sequence)
else:
    print("Không có số Fibonacci nào trong dãy số đã nhập.")

File: Bai_5/main.cpp
#include <iostream>
#include <vector>
using namespace std;

int fibonacci(int n) 
{
    if (n <= 1) 
    {
        return n;
    }
    return fibonacci(n - 1) + fibonacci(n - 2);
}

vector<int> findFibonacciInSequence(vector<int> sequence) 
{
    vector<int> fibonacciNumbers;
    for (int num : sequence) 
    {
        int i = 0;
        while (fibonacci(i) <= num) 
        {
            if (fibonacci(i) == num) 
            {
                fibonacciNumbers.push_back(num);
                break;
            }
            i++;
        }
    }
    return fibonacciNumbers;
}

int main() 
{
    cout << "Nhap vao mot day so: ";
    vector<int> numbers;
    int input;
    while (cin >> input) 
    {
        numbers.push_back(input);
        if (cin.peek() == '\n') 
        {
            break;
        }
    }

    vector<int> fibonacciInSequence = findFibonacciInSequence(numbers);
    if (!fibonacciInSequence.empty()) 
    {
        cout << "Cac so Fibonacci trong day so da nhap la:" << endl;
        for (int num : fibonacciInSequence) 
        {
            cout << num << " ";
        }
        cout << endl;
    } 
    else 
    {
        cout << "Khong co so Fibonacci nao trong day so da nhap." << endl;
    }

    return 0;
}
